cmake/serialConfig.cmake

```diff
@@ -0,0 +1,3 @@
+get_filename_component(SERIAL_CMAKE_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)
+set(SERIAL_INCLUDE_DIRS "${SERIAL_CMAKE_DIR}/../../../include")
+find_library(SERIAL_LIBRARIES serial PATHS ${SERIAL_CMAKE_DIR}/../../../lib/serial)
```

Comment:
If we're adding a config module, we should write a config module that exports a target instead of doing the old school approach of setting some variables that downstream projects have to use. that means we need to create an export set for the library target and install that export set then this file becomes a one liner that will look something like

```cmake
include(${CMAKE_CURRENT_LIST_DIR}/serial-targets.cmake)
```

Reply:
@ChrisThrasher can you point to helpful documentation and examples on how to do this well?

Reply:
https://github.com/ChrisThrasher/argon/blob/9a2982c5471f06ca1d97f8d2d8a1aab3055ba456/CMakeLists.txt#L29-L42

I'm biased but this installation code I wrote I quite like. It's goes above and beyond to be as robust and idiomatic as possible. You don't need to copy everything it does but it should do a good job laying our the basic steps.

1. Install all public headers. I do this by installing the entire `include/` directory.
2. Create export set. This only includes a single library target. It uses variables from the GNUInstallDirs module to provide a platform-agnostic installation that matches the conventions of whatever platform a user is using. I include the package name and version in the install tree but you can omit that.
3. Install said export set. This is where you can (and should) specify a namespace for all exported targets.
4. Create a version config file. This ensures that users can ask for a specific version and that will be checked against the version that is installed.
5. Install the config module and the version config file.

Reply:
> instead of doing the old school approach of setting some variables that downstream projects have to use.

This is not old-school. The normal usage pattern would be something like below, assuming that the serial library lives in a subdirectory

    include(libs/serial)
    target_link_libraries(${PROJECT_NAME} PRIVATE wjserial::serial)

This would just require a one-liner in CMakeLists.txt defining a ALIAS target. It relies on cmake's support for transitive dependencies. Adding such a target would simplify documentation and overall usage a lot.

Reply:
Yes, that's what I'm saying. We're in agreement.

Reply:
> It aids in readability and grepability if you simply spell out the variable name.

This is actually a personal preference... I actually agree on the personal level, but let's face it: the use of _${PROJECT_NAME}_ is    common enough in the cmake community to leave in place.

Reply:
Don't mistake commonality for it being a good idea. Lots of bad practices are still popular but that should not be used as an argument in favor of perpetuating bad ideas.